ext/nnstreamer/tensor_filter/tensor_filter_python.cc

```diff
@@ -630,7 +643,11 @@ PYCore::run (const GstTensorMemory * input, GstTensorMemory * output)
 
   PyObject *result = PyObject_CallMethod(core_obj, (char*) "invoke", (char*) "(O)", param);
   if (result) {
-    g_assert((unsigned int) PyList_Size(result) == outputTensorMeta.num_tensors);
+    if ((unsigned int) PyList_Size(result) != outputTensorMeta.num_tensors) {
+      res = -EINVAL;
+      ml_logf ("The Python allocated size mismatched. Cannot proceed.\n");
+      goto exit_decref;
```

Comment:
Before going to `exit_decref`, you need to call `Py_XDECREF (result)`

Reply:
Got it.